src/test/scala/com/fulcrumgenomics/umi/ExtractUmisFromBamTest.scala

```diff
@@ -0,0 +1,177 @@
+/*
+ * The MIT License
+ *
+ * Copyright (c) 2016 Fulcrum Genomics
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ *
+ */
+
+package com.fulcrumgenomics.umi
+
+import com.fulcrumgenomics.testing.{SamRecordSetBuilder, UnitSpec}
+import com.fulcrumgenomics.util.ReadStructure
+import dagr.sopt.cmdline.ValidationException
+import htsjdk.samtools.{SAMFileWriterFactory, SAMRecord}
+
+/**
+  * Tests for ExtractUmisFromBam.
+  */
+class ExtractUmisFromBamTest extends UnitSpec {
+
+  def newBam = makeTempFile("extract_umis_from_bam_test.", ".bam")
+
+  def annotateRecordFragment: SAMRecord = {
```

Comment:
This could be simplified to:
```scala
new SamRecordSetBuilder(readLength = 100).addFrag(name = "Frag", start = 1).map(_.setReadString("A" * 100)).head
```